libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,499 @@
+                                        vcObject.object(
+                                                "metadata",
+                                                metadata -> {
+                                                    metadata.stringType("reason", "testing");
+                                                    metadata.stringType(
+                                                            "timestampMs", "1711721297123");
+                                                    metadata.stringType(
+                                                            "txmaEventId",
+                                                            "1a116fe7-2ff9-4f7c-940d-d55fa7d03d66");
+                                                });
+                                        vcObject.stringType("provenance", "ONLINE");
+                                    });
+                        })
+                .build();
+    }
+
+    private DslPart getResponseBodyCreateUserVC() {
+        return newJsonBody(
+                        body -> {
+                            body.stringType("messageId", "bd8359d9-d559-47dd-9467-2a31e88a9e2d");
+                        })
+                .build();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "validCreateUserVcReturnsMessageIdWith200")
+    void testCreateVcRequestReturnsUsersVcWith200(MockServer mockServer) throws Exception {
+        // Under Test
+        EvcsClient evcsClient = new EvcsClient(mockConfigService);
+        try {
+            evcsClient.storeUserVCs(TEST_USER_ID, EVCS_CREATE_USER_VCS_DTO);
+        } catch (EvcsServiceException e) {
+            fail("EvcsServiceException was thrown");
+        }
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact invalidCreateUserVcReturns400(PactDslWithProvider builder) {
+        return builder.given("EVCS client exist")
+                .given("User has a valid VC")
+                .uponReceiving("A request to create EVCS user VCs")
+                .path("/vcs/" + INVALID_USER_ID)
+                .method("POST")
+                .headers(Map.of(CONTENT_TYPE, ContentType.APPLICATION_JSON.toString()))
+                .body(getRequestBodyForUserVC())
+                .willRespondWith()
+                .status(400)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "invalidCreateUserVcReturns400")
+    void testCreateUserVcRequestReturns400(MockServer mockServer) {
+        // Under Test
+        EvcsClient evcsClient = new EvcsClient(mockConfigService);
+        assertThrows(
+                EvcsServiceException.class,
+                () -> {
+                    evcsClient.storeUserVCs(INVALID_USER_ID, EVCS_CREATE_USER_VCS_DTO);
+                });
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact validCreateUserVcReturns409(PactDslWithProvider builder) {
```

Comment:
This could be a bit more specific - perhaps something like `createUserVcForExistingVcReturns409`?